src/TextForEvent.js

```diff
@@ -248,6 +248,26 @@ function textForPowerEvent(event) {
     });
 }
 
+function textForWidgetEvent(event) {
+    const senderName = event.sender ? event.sender.name : event.getSender();
+    const previousContent = event.getPrevContent() ? event.getPrevContent() : {};
+    const {name, type, url} = event.getContent() ? event.getContent() : {};
+    let widgetName = widgetName || name || type || previousContent.type || '';
+
+    // Apply sentence case
+    widgetName = widgetName ? widgetName[0].toUpperCase() + widgetName.slice(1).toLowerCase() + ' ' : '';
+
+    if (url) {
```

Comment:
Technically if the widget state event has no content, then it has been removed, but I think this is sufficiently equivalent as having a url is required.